.github/workflows/typescript_coverage.yml

```diff
@@ -0,0 +1,17 @@
+name: Typescript Coverage
+
+on:
+  push:
+    branches: ["**"]
+
+jobs:
+  coverage:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - name: Install dependencies and run code coverage
+        run: npm ci && npm run coverage
```

Comment:
what does npm ci do?

Reply:
@pogi7 npm ci (clean install) deletes all existing `node_modules` in the repo (on the test instance) and re-installs them with the exact versions specified in `package-lock.json`. This is recommended for automated testing environments (like CICD via GH actions), since it is more deterministic than using `package.json`, which can vary in terms of minor/patch version numbers between different development environments.

Here is where I read this info:
https://docs.npmjs.com/cli/v10/commands/npm-ci
https://stackoverflow.com/questions/52499617/what-is-the-difference-between-npm-install-and-npm-ci
https://www.the-koi.com/projects/setting-up-a-superfast-ci-with-vitest-and-github-actions/